gapis/api/vulkan/transform_overdraw.go

```diff
@@ -484,14 +484,26 @@ func (overdrawTransform *stencilOverdraw) createRenderPass(ctx context.Context,
 		return res
 	}
 
-	attachments := rpInfo.AttachmentDescriptions().All()
-	newAttachments := rpInfo.AttachmentDescriptions().Clone(api.CloneContext{})
-	newAttachments.Add(uint32(newAttachments.Len()), stencilAttachment)
-	newAttachmentsData, newAttachmentsLen := unpackMapWithAllocator(allocAndRead,
-		newAttachments)
+	newAttachments := []VkAttachmentDescription{}
+	for _, k := range rpInfo.AttachmentDescriptions().Keys() {
+		ad := rpInfo.AttachmentDescriptions().Get(k)
+		newAttachments = append(newAttachments, NewVkAttachmentDescription(
+			ad.Flags(),
+			ad.Fmt(),
+			ad.Samples(),
+			ad.LoadOp(),
+			ad.StoreOp(),
+			ad.StencilLoadOp(),
+			ad.StencilStoreOp(),
+			ad.InitialLayout(),
+			ad.FinalLayout(),
+		))
+	}
+	newAttachments = append(newAttachments, stencilAttachment)
+	newAttachmentsPtr := allocAndRead(newAttachments).Ptr()
 
 	stencilAttachmentReference := NewVkAttachmentReference(
-		uint32(len(attachments)),
+		uint32(len(newAttachments)-1),
```

Comment:
Was the lack of a -1 before a bug? Or is adding the -1 now a bug? Not sure if this is meant to be the number of elements in the array or the index of the last element.

Reply:
That's index of an element.

Difference is that previously, existing attachments would be copied to a local variable(`attachments`) and currently, we do not do that and copy to the `newAttachments` directly instead while iterating over them. And after the iteration, we add the new attachment to that list as well. that's why the -1 exists.

Reply:
Worth a comment on why we're skipping the last element then!